cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/DependenciesHelper.java

```diff
@@ -172,12 +172,19 @@
         return result;
     }
 
-    private Set<String> getSystemPropertiesFromLoopStatement(Map<String, Serializable> preStepActionData) {
+    private Set<String> getSystemPropertiesFromLoopAndWorkerGroupStatement(
+            Map<String, Serializable> preStepActionData) {
         Set<String> result = new HashSet<>();
         for (Map.Entry<String, Serializable> entry : preStepActionData.entrySet()) {
             if (entry.getValue() instanceof LoopStatement) {
                 result.addAll(((LoopStatement) entry.getValue()).getSystemPropertyDependencies());
             }
+            if (entry.getValue() instanceof WorkerGroupStatement) {
```

Comment:
Lets refactor a bit this for and use a lambda filter and extract a few methods for the checks if necessary